### ISSUE TITLE
GeoSwift Crashes

### ISSUE THREAD

**Issue by macdrevx** [GEOSwift/GEOSwift#85]
The following code snippet crashes in GeoSwift (tested in 2.1.0, 2.1.3, swift 3.2, swift 4.0):

```
let jsonString = "{\"type\":\"MultiPolygon\",\"coordinates\":[[[[-78.926467895507812,36.051319575025502],[-78.852289148724822,36.033005280682119],[-78.791885375976562,35.996896453738117],[-78.734207153320312,35.951329861522673],[-78.640823364257812,35.927980690382697],[-78.481521606445312,35.885712278670248],[-78.452682495117188,35.74316846139925],[-78.50833730594735,35.68130132015127],[-78.613357543945312,35.670685013302347],[-78.714981079101562,35.684071533140973],[-78.804244995117188,35.679609609368583],[-78.874276302296835,35.693002873745471],[-78.911361694335938,35.736480443001462],[-78.931961059570312,35.823380847706538],[-79.003372192382812,35.931316708569028],[-78.988265991210938,36.022446681758488],[-78.926467895507812,36.051319575025502]]]]}"
let jsonData = jsonString.data(using: .utf8)!
let json = try! JSONSerialization.jsonObject(with: jsonData, options: [])

guard let dictionary = json as? [String: AnyObject],
    let features = Geometry.fromGeoJSONDictionary(dictionary) else {
        fatalError()
}

let points = (features[0]
    .geometries![0]
    .boundary() as! GeometryCollection<LineString>)
    .geometries[0]
    .points
```

The crash doesn't always happen in the same place. Happy to provide more info as needed.

**Comment by raugturi** [comment]
If it helps, the problem appears to be resolved if the boundary() call is not chained with anything else. I did receive this error at one point: "libc++abi.dylib: Pure virtual function called!" which leads me to believe some part of this chaining is calling a virtual function inside of a constructor or destructor. I just don't know how that's the cause or how to find it.

I did find this stackoverflow about it here: https://stackoverflow.com/questions/34051749/how-to-track-down-libcabi-dylib-pure-virtual-function-called-in-xcode

Will look into some of the suggestions to see if that leads to understanding what's happening.

**Comment by jwilson85** [comment]
Im getting a force unwrap crash on the line 
let boundaryGEOM = GEOSBoundary_r(GEOS_HANDLE, self.geometry)!
Inside the func boundary() -> Geometry {

**Comment by vfn** [comment]
@jwilson85, while unrelated to the OP's issue, this is still a problem.

Unfortunately all those function use force unwrapping, and it will result in a crash whenever GEOS returns nil.

We could either change this functions to return nil, in case of failure, or to make them to throw an error. Both changes would be breaking changes tho.

Would you like to have a go at it?

**Comment by vfn** [comment]
@jwilson85 I believe the library does have some more force unwrapping happening on other files. But as a starting point, try to work on the file your pointed out, and if you find anywhere else, please feel free to change as you go.

Thanks!

**Comment by jwilson85** [comment]
I have never done a git hub PR before Do I need access from you? @vfn

**Comment by vfn** [comment]
@jwilson85 that's awesome! Please create a pull request https://github.com/GEOSwift/GEOSwift/pulls and we will review your changes. Thanks!

**Comment by macdrevx** [comment]
@jwilson85 if you haven't already done so, you may want to create a fork of this repo. You can push your changes there and then create the PR back to the main repo.

**Comment by macdrevx** [comment]
@vfn @jwilson85: I'm spending a few days working with @raugturi on a fix for this issue. We'll have a PR ready today or Monday with more info. The fix necessarily breaks the API, so we'll need to increment the major version number before releasing it. We've come up with a number of other ideas for this project as well, some of which would constitute fairly major changes to the API. Would anyone be interested in setting up a public Slack or other channel for having some discussion about what direction to go in?

**Comment by jwilson85** [comment]
@macdrevx I hope my merged PR helped

**Comment by macdrevx** [comment]
This was fixed by #100. It will be released with v3.0.0 since it's a breaking change.
